core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -614,20 +621,23 @@ public void testCompleteCreateTable() {
             .create();
 
     // validate table settings
-    Assert.assertEquals(
-        "Table name should report its full name", catalog.name() + "." + ident, table.name());
-    Assert.assertTrue("Table should exist", catalog.tableExists(ident));
-    Assert.assertEquals(
-        "Schema should match expected ID assignment",
-        TABLE_SCHEMA.asStruct(),
-        table.schema().asStruct());
-    Assert.assertNotNull("Should have a location", table.location());
-    Assert.assertEquals("Should use requested partition spec", TABLE_SPEC, table.spec());
-    Assert.assertEquals("Should use requested write order", TABLE_WRITE_ORDER, table.sortOrder());
-    Assert.assertEquals(
-        "Table properties should be a superset of the requested properties",
-        properties.entrySet(),
-        Sets.intersection(properties.entrySet(), table.properties().entrySet()));
+    Assertions.assertThat(table.name())
+        .as("Table name should report its full name")
+        .isEqualTo(catalog.name() + "." + ident);
+    Assertions.assertThat(catalog.tableExists(ident)).as("Table should exist").isTrue();
+    Assertions.assertThat(table.schema().asStruct())
+        .as("Schema should match expected ID assignment")
+        .isEqualTo(TABLE_SCHEMA.asStruct());
+    Assertions.assertThat(table.location()).as("Should have a location").isNotNull();
+    Assertions.assertThat(table.spec())
+        .as("Should use requested partition spec")
+        .isEqualTo(TABLE_SPEC);
+    Assertions.assertThat(table.sortOrder())
+        .as("Should use requested write order")
+        .isEqualTo(TABLE_WRITE_ORDER);
+    Assertions.assertThat(table.properties().entrySet())
+        .as("Table properties should be a superset of the requested properties")
+        .containsAll(properties.entrySet());
```

Comment:
@skytin1004 can you make that change please so that we can get this in?

Reply:
@skytin1004 this still uses `containsAll` rather than `containsExactlyInAnyOrder`. We want to make sure that `table.properties()` doesn't contain any more elements, hence the `containsExactlyInAnyOrder` check